var/spack/repos/builtin/packages/ripgrep/package.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Ripgrep(Package):
```

Comment:
Hm, we don't have a `RustPackage` or `CargoPackage` yet?
Do you want to define one? :)

Reply:
I don't know enough about how it all works to deal with it yet; it suffers from the "competing package manager & how do we handle dependencies that cargo wants to handle" issue and I don't know what the solution would look like.

I still have finishing up support for Go (#14061) on the list (I'd love input from more than me and @scheibelp about whether the spack community trusts the Go module systems signatures or wants to Do It Ourselves).

Reply:
no prob, maybe @trws and @sknigh have an idea on this as they committed rust-related works to Spack before.

Reply:
I’d leave it for now, we have other rust packages that are built just like this.  There’s also a bit of a policy decision we need to think about when making a first-class cargo package class.  At the moment this, and others like it, fetch most of their actual deps with cargo and spack is completely unaware of them.  Either way, I wouldn’t hold this one up on it.